src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs

```diff
@@ -254,42 +256,49 @@ public async Task ProcessFinishedReleasesAsync(CancellationToken cancellationTok
                                 else
                                 {
                                     Logger.LogInformation($"Release {releaseId}, channel {channelId} finished executing");
+
+                                    if (release.Environments[0].Status == AzureDevOpsReleaseStatus.Rejected)
+                                    {
+                                        await CreateGitHubIssueAsync(buildId, releaseId, release.Name);
+                                    }
                                 }
                             }
-
-                            if (unfinishedReleases.Count > 0)
+                            catch (TaskCanceledException tcex) when (tcex.CancellationToken == cancellationToken)
                             {
-                                await runningPipelines.TryUpdateAsync(tx, buildId, unfinishedReleases, releaseStatuses);
+                                // ignore
                             }
-                            else
+                            catch (Exception ex)
                             {
-                                Logger.LogInformation($"All releases for build {buildId} for channel {channelId} finished. Creating BuildChannel.");
-
-                                buildChannelsToAdd.Add(new BuildChannel
-                                {
-                                    BuildId = buildId,
-                                    ChannelId = channelId
-                                });
-                                await runningPipelines.TryRemoveAsync(tx, buildId);
+                                // Something failed while fetching the release information so the potential created issue wouldn't have relevant information to
+                                // be notified so we just log the exception to AppInsights with not filed issue.
+                                Logger.LogError(ex, "Processing finished releases");
```

Comment:
I think we should fill an issue for this case. This might mean that we didn't even created a release for publishing, we need to be aware of that since if the execution got here it means the build was supposed to create a release + publish.

Reply:
Also, could you please change the text of LogError? Last time I debugged this the message was confusing.

Reply:
In the case where we were hitting 404s. This would mean we'll open an issue every 5 minutes. That's a no go

Reply:
You need to open the issue and remove the release from the pending list so we don't need to keep checking 404 things everytime.